samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -449,31 +449,25 @@ mfxStatus CDecodingPipeline::Init(sInputParams *pParams)
     }
 #endif
 
-    bool isDeviceRequired = false;
     mfxHandleType hdl_t;
 #if D3D_SURFACES_SUPPORT
-    isDeviceRequired = m_memType != SYSTEM_MEMORY || !m_bDecOutSysmem;
     hdl_t =
 #if MFX_D3D11_SUPPORT
-        D3D11_MEMORY == m_memType ? MFX_HANDLE_D3D11_DEVICE :
+        D3D9_MEMORY != m_memType ? MFX_HANDLE_D3D11_DEVICE :
 #endif // #if MFX_D3D11_SUPPORT
         MFX_HANDLE_D3D9_DEVICE_MANAGER;
 #elif LIBVA_SUPPORT
-    isDeviceRequired = true; // on Linux MediaSDK doesn't create device internally
     hdl_t = MFX_HANDLE_VA_DISPLAY;
 #endif
-    if (isDeviceRequired)
+    sts = CreateHWDevice();
+    MSDK_CHECK_STATUS(sts, "CreateHWDevice failed");
+    if (pParams->bUseHWLib)
     {
-        sts = CreateHWDevice();
-        MSDK_CHECK_STATUS(sts, "CreateHWDevice failed");
-        if (pParams->bUseHWLib)
-        {
-            mfxHDL hdl = NULL;
-            sts = m_hwdev->GetHandle(hdl_t, &hdl);
-            MSDK_CHECK_STATUS(sts, "m_hwdev->GetHandle failed");
-            sts = m_mfxSession.SetHandle(hdl_t, hdl);
-            MSDK_CHECK_STATUS(sts, "m_mfxSession.SetHandle failed");
-        }
+        mfxHDL hdl = NULL;
+        sts = m_hwdev->GetHandle(hdl_t, &hdl);
+        MSDK_CHECK_STATUS(sts, "m_hwdev->GetHandle failed");
+        sts = m_mfxSession.SetHandle(hdl_t, hdl);
+        MSDK_CHECK_STATUS(sts, "m_mfxSession.SetHandle failed");
```

Comment:
Seems itsn't required. With system memory:

- on Windows , MSDK can create device internally. But still, please make sure we set `impl |= MFX_IMPL_VIA_D3D11`
- On Linux we always need to pass device for all memory types. And it's okay to create system memory allocator

Reply:
Done.